uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Page/UserMenu.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+use CRM_HRCore_CMSData_PathsFactory as PathsFactory;
+
+class CRM_HRCore_Page_UserMenu extends CRM_Core_Page {
+
+  private $contactData;
+  private $cmsPaths;
+
+  public function run() {
+    $this->getContactData();
+    $this->instantiateCmsPaths();
+
+    $this->assign('username', $this->contactData['display_name']);
+    $this->assign('image', $this->getUserImagePath());
+    $this->assign('editLink', $this->cmsPaths->getEditAccountPath());
+    $this->assign('logoutLink', $this->cmsPaths->getLogoutPath());
+
+    return parent::run();
+  }
+
+  /**
+   * Gets the currently logged in contact's data, including the
+   * her user id in the CMS
+   *
+   * @return array
+   */
+  private function getContactData() {
+    $rawContactData = civicrm_api('Contact', 'getsingle', array(
```

Comment:
getsingle will through an exception if it found nothing, so I would either catch the exception if it happen, but this will add make the code more nested so I would choose get instead of getsingle and add limit => 1 (but the output will be a bit different)

also please use [] instead of array().

also why not to use civicrm_api3 instead of passing the version.